pkgs/development/tools/yarn2nix-moretea/yarn2nix/default.nix

```diff
@@ -99,9 +99,12 @@ in rec {
           ""
       ) (builtins.attrNames pkgConfig));
 
+      package = lib.importJSON packageJSON;
+      workspacePackageResolutions = (package.resolutions or {}) // packageResolutions;
+
       workspaceJSON = pkgs.writeText
         "${name}-workspace-package.json"
-        (builtins.toJSON { private = true; workspaces = ["deps/**"]; resolutions = packageResolutions; }); # scoped packages need second splat
+        (builtins.toJSON { private = true; workspaces = ["deps/**"]; resolutions = workspacePackageResolutions; }); # scoped packages need second splat
```

Comment:
Here's an example implementation of @Radvendii's idea (build-time JSON generation) to avoid IFD:

```nix
      workspaceJSON = pkgs.runCommand "${name}-workspace-package.json" {
		nativeBuildInputs = [ pkgs.jq ];
		inherit packageJSON;
		passAsFile = [ "baseJSON" ];
		baseJSON = builtins.toJSON { private = true; workspaces = ["deps/**"]; resolutions = packageResolutions; }; # scoped packages need second splat
	  } ''
	    jq --slurpfile packageJSON "$packageJSON" '.resolutions = $packageJSON[0].resolutions + .resolutions <"$baseJSONPath"
	  '';
```

Reply:
That works 🎉